### PR TITLE
Add oscalkit-test

### PR DESCRIPTION
**Pull request by farhan-khan30**
Signed-off-by: farhankamalkhan <farhankamalkhan@gmail.com>

### CODE REVIEW

File: oscalkit-test/methods.go [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+
+	_, err = io.Copy(catalog, data.Body)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// ProfileParcing method to parse the profile and return the controls and subcontrols
+func ProfileParcing(parsedProfile *profile.Profile) []string {
```
Comment (mohuk) [inline oscalkit-test/methods.go]: Typo: should be `ProfileParsing`

File: oscalkit-test/methods.go [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+		save := strings.Split(parsedProfile.Imports[l].Href.Path, "/")
+		filename := "./tmp/" + save[len(save)-1]
+		f, err := os.Open(filename)
+		if err != nil {
+			log.Fatal(err)
+		}
+		check, _ := ProfileCatalogCheck(f)
+		if check == "Catalog" {
+
+			ProfileControls := ImportParcing(parsedProfile, parsedProfile.Imports[l].Href.Path)
```
Comment (mohuk) [inline oscalkit-test/methods.go]: Typo: should be `ImportParsing`

File: oscalkit-test/methods.go [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+				log.Fatal(err)
+			}
+
+			parsedProfile1, err := GetProfile(f)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			save := ProfileProcessing(parsedProfile1)
+			save1 := ImportParcing(parsedProfile, parsedProfile.Imports[l].Href.Path)
```
Comment (mohuk) [inline oscalkit-test/methods.go]: Typo: should be `ImportParsing`

File: oscalkit-test/methods.go [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+		if count > 0 {
+			result = append(result, s2element)
+		}
+		count = 0
+	}
+	return result
+}
+
+// ImportParcing method to parse the profile and return the controls and subcontrols
+func ImportParcing(parsedProfile *profile.Profile, link string) []string {
```
Comment (mohuk) [inline oscalkit-test/methods.go]: Typo: should be `ImportParsing`

File: oscalkit-test/TestCases.go [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+	"os"
+
+	"github.com/fatih/color"
+	"github.com/opencontrol/oscalkit/types/oscal/catalog"
+)
+
+// SecurityControlsSubcontrolCheck is a test to verify that all controls from the catalog are being mapped correctly
+func SecurityControlsSubcontrolCheck(check []catalog.Catalog, ProfileFile string) error {
+
+	CodeGeneratedControls := ProtocolsMapping(check)
```
Comment (anweiss) [inline oscalkit-test/TestCases.go]: camel case

```suggestion
 	codeGeneratedControls := ProtocolsMapping(check)
```

File: oscalkit-test/TestCases.go [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	parsedProfile, err := GetProfile(f)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	ProfileControlsDetails := ProfileProcessing(parsedProfile)
```
Comment (anweiss) [inline oscalkit-test/TestCases.go]: camel case

```suggestion
 	profileControlsDetails := ProfileProcessing(parsedProfile)
```

File: oscalkit-test/TestCases.go [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+	if len(CodeGeneratedControls) == len(ProfileControlsDetails) {
+		println("Perfect Count Match")
+		println("Go file control, sub-control count: ", len(CodeGeneratedControls))
+		println("Profile control, sub-control count: ", len(ProfileControlsDetails))
+		CodeGeneratedMapping := ProtocolsMapping(check)
```
Comment (anweiss) [inline oscalkit-test/TestCases.go]: camel case

```suggestion
 		codeGeneratedMapping := ProtocolsMapping(check)
```

File: oscalkit-test/methods.go [hunk 8]
```diff
@@ -0,0 +1,322 @@
...
+			return true
+		}
+	}
+	return false
+}
+
+// DownloadCatalog writes the JSON of the provided URL into a catalog.json file
+func DownloadCatalog(url string) error {
+	save := strings.Split(url, "/")
+	err := os.Mkdir("./tmp", 0777)
```
Comment (anweiss) [inline oscalkit-test/methods.go]: Use the built-in function for creating a temporary directory. `ioutil.TempDir()`. See https://golang.org/pkg/io/ioutil/#TempDir

**Comment by farhan-khan30** [comment]
@anweiss @mohuk The changes requested have been made.

**Comment by farhan-khan30** [comment]
@anweiss Just as a clarification, we are using github.com/fatih/color to display the test result in green or red in case the test passes or fails respectively, currently the user will have to go get the dependency and this is mentioned in the readme file. Is this approach correct or should we add it to vendor?

**Comment by anweiss** [comment]
@farhan-khan30 please vendor the deps

**Comment by anweiss** [comment]
@farhan-khan30 also bear in mind that https://github.com/fatih/color was recently archived and is no longer maintained.

**Comment by anweiss** [comment]
LGTM.

**Pull request merged**